src/register/addresses.rs

```diff
@@ -0,0 +1,678 @@
+//! Addresses for CSRs. Automatically generated by parse_opcodes.
```

Comment:
A link to how this was generated would be nice.

Reply:
I looked into `parse_opcodes`, but it doesn't actually seem like that is capable of generating CSR information. The official riscv-opcodes library does so I can see about using that to do the generation instead.